feat: Add isGasFeeIncluded property (#6428)
MetaMask/core: packages/transaction-controller/CHANGELOG.md

## [Unreleased]

### Added

> - Add `isGasFeeIncluded` to `TransactionMeta`, `TransactionBatchRequest` and `addTransaction` options so that the client can signal that the gas fee is included in the transaction. ([#6428](https://github.com/MetaMask/core/pull/6428))

Reviewer: ```suggestion
- Add `isGasFeeIncluded` to `TransactionMeta`, `TransactionBatchRequest` and `addTransaction` options so the client can signal that MetaMask is compensated for the gas fee by the transaction ([#6428](https://github.com/MetaMask/core/pull/6428))
```
